drivers/node/node_registry.go

```diff
@@ -0,0 +1,50 @@
+package node
+
+import (
+	"fmt"
+	"os/exec"
+)
+
+var nodeRegistry = make(map[string]Node)
+
+// AddNode adds a node to the node collection
+func AddNode(n Node) error {
+	if n.uuid != "" {
+		return fmt.Errorf("UUID should not be set to add new node")
+	}
+	uuid, err := exec.Command("uuidgen").Output()
```

Comment:
instead of exec, you could use a golang library to generate UUID.
github.com/pborman/uuid